### PR TITLE
Issue #1429 split hyperv checks into specific functions.

### PR DESCRIPTION
**Pull request by anjannath**

### CODE REVIEW

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -61,8 +61,20 @@ func preflightChecksBeforeStartingHost() {
...
-			checkHypervDriver,
-			"Checking if Hyper-V driver is configured",
+			checkHypervDriverInstalled,
+			"Checking if Hyper-V driver is installed",
+			false, configCmd.WarnCheckHyperVDriver.Name,
+			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckHyperVDriver.Name,
+			checkHypervDriverSwitch,
+			"Checking if Hyper-V driver is configured (Virtual Switch)",
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: This should be `Checking if Hyper-V driver is configured with Virtual Switch`

File: cmd/minishift/cmd/start_preflight.go [hunk 2]
```diff
@@ -61,8 +61,20 @@ func preflightChecksBeforeStartingHost() {
...
+			false, configCmd.WarnCheckHyperVDriver.Name,
+			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckHyperVDriver.Name,
+			checkHypervDriverUser,
+			"Checking if Hyper-V driver is configured (HyperV Administrators)",
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: `Checking if user is member of HyperV Administrators`

File: cmd/minishift/cmd/start_preflight.go [hunk 3]
```diff
@@ -231,23 +243,34 @@ func checkKvmDriver() bool {
...
+func checkHypervDriverSwitch() bool {
 	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
 	if switchEnv == "" {
 		return false
 	}
 
+	return true
+}
+
+//checkHypervDriverInstalled returns ture if Hyper V driver is installed
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: Space after `//` and hyphen in `Hyper-V`

File: cmd/minishift/cmd/start_preflight.go [hunk 4]
```diff
@@ -231,23 +243,34 @@ func checkKvmDriver() bool {
...
 
 	checkIfHyperVInstalled := `@(Get-Command Get-VM).ModuleName`
 	stdOut, _ := posh.Execute(checkIfHyperVInstalled)
 	if !strings.Contains(stdOut, "Hyper-V") {
 		return false
 	}
+	return true
+}
+
+//checkHypervDriverUser returns true if user is member of Hyper V admin
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: Space after // and hyphen in Hyper-V

**Comment by anjannath** [comment]
@budhrg Made the changes.

File: cmd/minishift/cmd/start_preflight.go [hunk 5]
```diff
@@ -59,8 +61,20 @@ func preflightChecksBeforeStartingHost() {
...
-			checkHypervDriver,
-			"Checking if Hyper-V driver is configured",
+			checkHypervDriverInstalled,
+			"Checking if Hyper-V driver is installed",
+			false, configCmd.WarnCheckHyperVDriver.Name,
+			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckHyperVDriver.Name,
+			checkHypervDriverSwitch,
+			"Checking if Hyper-V driver is configured with Virtual Switch",
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: Forgot to mention, I think it should be more explicit.   ` ... External Virtual Switch`. See https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html#hyper-v-driver

Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: Forgot to mention, I think it should be more explicit.   ` ... External Virtual Switch`. See https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html#hyper-v-driver

File: cmd/minishift/cmd/start_preflight.go [hunk 6]
```diff
@@ -59,8 +61,20 @@ func preflightChecksBeforeStartingHost() {
...
+			false, configCmd.WarnCheckHyperVDriver.Name,
+			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckHyperVDriver.Name,
+			checkHypervDriverUser,
+			"Checking if user is member of Hyper-V Administrators",
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: Please update this also, `... Hyper-V Administrator group`

File: cmd/minishift/cmd/start_preflight.go (hunk 3 above)
Comment (gbraad) [reply cmd/minishift/cmd/start_preflight.go]: `ture` => `true`

**Comment by gbraad** [comment]
Try to use Hyper-V when describing the hypervisor in commit messages and descriptions, but use hyperv of HyperV in code

**Comment by anjannath** [comment]
@budhrg @gbraad Made the mentioned changes.

**Comment by gbraad** [comment]
Merged as af7bffc54d5e030af37a38419756dc82e2a159a0

**Pull request closed without merging**